src/context.rs

```diff
@@ -183,6 +191,14 @@ impl<'gcc, 'tcx> CodegenCx<'gcc, 'tcx> {
         // TODO(antoyo): set alignment on those types as well.
         let float_type = context.new_type::<f32>();
         let double_type = context.new_type::<f64>();
+        #[cfg(feature = "master")]
+        let float16_type = context.new_c_type(CType::Float16);
+        #[cfg(feature = "master")]
+        let float32_type = context.new_c_type(CType::Float32);
+        #[cfg(feature = "master")]
+        let float64_type = context.new_c_type(CType::Float64);
+        #[cfg(feature = "master")]
+        let float128_type = context.new_c_type(CType::Float128);
```

Comment:
Those types might not be available on some platforms, so it might be wrong to query these types here.

Depending on how much you want to add stuff in this PR, you have two options:

 * Do the same as [detecting 128-bit integers support](https://github.com/antoyo/rustc_codegen_gcc/blob/master/src/lib.rs#L168-L170): we might want to change the [GCC API](https://github.com/rust-lang/gcc/blob/master/gcc/jit/libgccjit.cc#L4010) to take a type as a parameter.
 * Only query the types in the functions like `type_f16` below so that it only panics when used.

Reply:
So far I query the types only in the functions().
As next step I will try to detect if new types are support.

Reply:
I mean, just calling `new_c_type()` here will give an error if the type is not supported.
[Here's how I detect 128-bit support with the old way of doing things.](https://github.com/antoyo/rustc_codegen_gcc/blob/master/src/lib.rs#L209-L220)